invenio_rdm_records/github/release.py

```diff
@@ -0,0 +1,315 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""Github release API implementation."""
+
+import requests
+import yaml
+from flask import current_app
+from invenio_github.api import GitHubRelease
+from invenio_github.models import ReleaseStatus
+from marshmallow import ValidationError
+from mistune import markdown
+from nameparser import HumanName
+
+from invenio_access.permissions import system_identity
+from invenio_db import db
+from invenio_pidstore.models import PersistentIdentifier
+from invenio_rdm_records.github.utils import get_extra_metadata, load_citation_metadata
+from invenio_rdm_records.proxies import current_rdm_records_service
+from invenio_records_resources.services.uow import UnitOfWork
+
+
+class RDMGithubRelease(GitHubRelease):
+    """Implement release API instance for RDM."""
+
+    def _related_identifiers(self):
```

Comment:
all the `_` method could be an API class with properties that extract from the actual GH release object, becoming a wapper API class with RDM-Records specific needs.